crates/cargo-util-schemas/src/core/package_id_spec.rs

```diff
@@ -277,6 +268,31 @@ impl<'de> de::Deserialize<'de> for PackageIdSpec {
     }
 }
 
+/// Error parsing a [`PackageIdSpec`].
+#[derive(Debug, thiserror::Error)]
+pub enum PackageIdSpecError {
```

Comment:
In exposing this enum, we are exposing our implementation details and we are very limited in what error changes we can make without breaking compatibility.